src/core/ddsc/src/dds_reader.c

```diff
@@ -482,15 +482,19 @@ static bool dds_reader_support_shm(const struct ddsi_config* cfg, const dds_qos_
     return false;
   }
 
-  uint32_t sub_history_req = cfg->sub_history_request;
+  // uint32_t sub_history_req = cfg->sub_history_request;
+  // MAKI: need the max from iceoryx here (available on master in the API but
+  // not in release_1.0)
+  const int32_t max_sub_history_req = 16;
```

Comment:
Perhaps it is possible to use `#if ...` to check whether the Iceoryx version used is new enough to provide the max value?

Reply:
I see no way to check for the version with the headers we currently have in `iceoryx_binding_c`. A header that allows something like this exists for the C++ library, `iceoryx_versions.hpp` but not the C library. I will create an issue for this in the next release but then this point is moot since the values are available with the API.